src/main.rs

```diff
@@ -29,6 +39,79 @@ macro_rules! upgrade_weak {
     };
 }
 
+macro_rules! save_settings {
+    ($x:ident, $call:ident, $($to_downgrade:ident),* => move |$($p:tt),*| $body:expr) => {{
+        $( let $to_downgrade = $to_downgrade.downgrade(); )*
+        $x.$call(move |$($p),*| {
+            $( let $to_downgrade = upgrade_weak!($to_downgrade, ()); )*
+            $body
+        });
+    }}
+}
+
+fn get_settings_file_path() -> PathBuf {
+    let mut path = glib::get_user_config_dir().unwrap_or_else(|| PathBuf::from("."));
+    path.push(APPLICATION_NAME);
+    path.push("settings.toml");
+    path
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Serialize, Deserialize)]
+enum OutputFormat {
+    JPEG,
+    PNG,
+}
+
+impl<'a> From<&'a str> for OutputFormat {
+    fn from(s: &'a str) -> Self {
+        match s.to_lowercase().as_str() {
+            "jpeg" => OutputFormat::JPEG,
+            "png" => OutputFormat::PNG,
+            _ => panic!("unsupported output format"),
+        }
+    }
+}
+
+impl From<Option<String>> for OutputFormat {
+    fn from(s: Option<String>) -> Self {
+        if let Some(s) = s {
+            match s.to_lowercase().as_str() {
+                "jpeg" => OutputFormat::JPEG,
+                "png" => OutputFormat::PNG,
+                _ => panic!("unsupported output format"),
+            }
+        } else {
+            OutputFormat::default()
+        }
+    }
+}
+
+impl Default for OutputFormat {
+    fn default() -> Self {
+        OutputFormat::JPEG
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+struct SnapshotSettings {
+    /// By default, the current one.
+    folder: PathBuf,
+    /// Format in which to save the snapshot.
+    format: OutputFormat,
+    /// Timer length in seconds.
+    timer_length: usize,
+}
+
+impl Default for SnapshotSettings {
+    fn default() -> SnapshotSettings {
+        SnapshotSettings {
+            folder: glib::get_home_dir().unwrap_or_else(|| PathBuf::from(".")),
+            format: OutputFormat::default(),
+            timer_length: 3,
```

Comment:
For some reason the timer length in the settings window is always shown as 0 for me and the +/- buttons don't work either